daemon/protocol/python/scoot/daemon_cli.py

```diff
@@ -0,0 +1,156 @@
+
+
+if __name__ == '__main__':
+    try:
+        cmd = docopt.docopt(__doc__)
+    except Exception as e:
+        print("exception:{0}".format(e))
+        raise docopt.DocoptExit(e)
+    try:
+#         print("starting client")
+        proto.start()
+    except proto.ScootException as e:
+        if re.search("UNAVAILABLE", e) > 0:
+            raise docopt.DocoptExit("Scoot Daemon is not running!\n")
+        raise docopt.DocoptExit(e)          
+        
+    if cmd['snapshot'] and cmd['create']:
+        if not cmd['<srcDir>']:
+            raise docopt.DocoptExit("Must provide srcDir!")
+        try:
+            sid = proto.create_snapshot(cmd['<srcDir>'])
+#             print("createSnapshot({0})".format(cmd['<srcDir>']))
+            print("snapshot id = {0}".format(sid))
+        except proto.ScootException as e:
+            if re.search("Not started", e) > 0 or re.search("UNAVAILABLE", e) > 0:
+                raise docopt.DocoptExit("Scoot Daemon is not running!\n")
+            raise docopt.DocoptExit(e)          
+        
+    elif cmd['checkout']:
+        if not cmd['<snapshotId>']:
+            raise docopt.DocoptExit("Must supply a snapshot id.")
+        if not cmd["<destDir>"]:
+            raise docopt.DocoptExit("Must supply a destination directory.")
+        try:
+            r = proto.checkout_snapshot(snapshot_id=cmd['<snapshotId>'], dirpath=cmd["<destDir>"])
+#             print("checkout({0}, {1})".format(cmd['<snapshotId>'], dirpath=cmd["<destDir>"]))
+        except Exception as e:
+            raise docopt.DocoptExit(e)
+        
+    elif cmd['run']:
+        timeout = cmd['--timeout']
+        if not timeout:
+            timeout = 500000
+        else:
+            try:
+                timeout = int(timeout)
+            except Exception as e:
+                raise docopt.DocoptExit("invalid value for timeout, must be an integer.")
+        if not cmd["--snapshotId"]:
+            raise docopt.DocoptExit("SnapshotId is missing")
+        args = re.split(" ", cmd['<command>'])  #TODO - fix this - need to handle spaces in \"
+
+        try:
+            runId = proto.run(snapshot_id=cmd["--snapshotId"], argv=args, timeout_ns=timeout)
+#             print("run({0}, {1}, {2})".format(["--snapshotId"], cmd["<command>"], timeout))
+            print("run id = {0}".format(runId))
+        except proto.ScootException as e:
+            if re.search("No resources available", e) is None:
+                print(e)
+            elif re.search("Not started", e) > 0 or re.search("UNAVAILABLE", e) > 0:
+                raise docopt.DocoptExit("Scoot Daemon is not running!\n")
+            raise docopt.DocoptExit(e)
+        except Exception as e1:
+            raise docopt.DocoptExit(e1)
+        
+    elif cmd['poll']:
+        wait = cmd["--wait"]
+        if not wait:
+            wait = 0
+        runIds = cmd["<runIds>"]
+        if not cmd["<runIds>"]:
+            raise docopt.DocoptExit("RunId list is empty!")
```

Comment:
We have an awful lot of code here devoted to manual parsing and checking input options.  If there is not a way to handle the subcommand and required parameters automatically w/ docopt I am going to suggest using the standard argparse library.